w3ctrace/state.go

```diff
@@ -19,13 +19,29 @@ type State []string
 func ParseState(traceStateValue string) (State, error) {
 	var state State
 
-	for _, st := range strings.SplitN(traceStateValue, ",", 32) {
-		st = strings.TrimSpace(st)
-		if st == "" {
-			continue
+	rawEntries := strings.Split(traceStateValue, ",")
+	rawEntriesLen := len(rawEntries)
+	if rawEntriesLen > 32 {
+		filtered := 0
+		for _, st := range rawEntries {
+			if rawEntriesLen-filtered <= 32 || len(state) == 32 {
+				break
+			}
+
+			if len(st) > 128 || st == "" {
+				filtered++
+				continue
+			}
+
+			state = append(state, st)
+		}
+	} else {
+		for _, st := range rawEntries {
+			if st == "" {
+				continue
+			}
```

Comment:
As a suggestion, if we stop filtering empty list items for each request, we could spare some CPU cycles and use a guard-clause here instead of `else`

Reply:
Like this https://github.com/instana/go-sensor/pull/287/files#diff-8e3fa4ec5050520f1e8157ac884ce463ca2553109301a7e640e2e0a5f9927aa7R38?